src/main/di.ts

```diff
@@ -156,6 +157,34 @@ if (!fs.existsSync(publicationRepositoryPath)) {
 // end of create database
 //
 
+// https://pouchdb.com/guides/compact-and-destroy.html
+const compactDb = async () => {
+    const res = await PromiseAllSettled([
+        publicationDb.compact(),
+        opdsDb.compact(),
+        configDb.compact(),
+        locatorDb.compact(),
+        lcpSecretDb.compact(),
+    ]);
+
+    const done = res.reduce((pv, cv) => pv && cv.status === "fulfilled", true);
+    if (!done) {
+        throw JSON.stringify(res);
+    }
+};
+
+const closeProcessLock = (() => {
+    let lock = false;
+
+    return {
+        get isLock() {
+            return lock;
+        },
+        lock: () => lock = true,
+        release: () => lock = false,
```

Comment:
there can be race conditions on the lock/release cycles, so instead I would suggest implementing a mutex for the critical section based on an incremental lock index, for example https://github.com/mgtitimoli/await-mutex/blob/master/src/mutex.js

...or an abstraction with try/finally to ensure the lock is released:
```
export class Mutex {
  private mutex = Promise.resolve();

  lock(): PromiseLike<() => void> {
    let begin: (unlock: () => void) => void = unlock => {};

    this.mutex = this.mutex.then(() => {
      return new Promise(begin);
    });

    return new Promise(res => {
      begin = res;
    });
  }

  async dispatch(fn: (() => T) | (() => PromiseLike<T>)): Promise<T> {
    const unlock = await this.lock();
    try {
      return await Promise.resolve(fn());
    } finally {
      unlock();
    }
  }
}
```
see full article source: https://spin.atomicobject.com/2018/09/10/javascript-concurrency/

Reply:
or if you want to go crazy on process reentrance, etc.
https://github.com/rogierschouten/async-lock/blob/master/lib/index.js
(this one is an NPM package)

Reply:
https://github.com/edrlab/thorium-reader/pull/1094/commits/36997799e397fd00c5398eb18117f2d88585266e